spec/requests/api/v0/user_requests/update_spec.rb

```diff
@@ -0,0 +1,45 @@
+require "rails_helper"
+
+RSpec.describe "User Update Profile Endpoint" do
+  describe "happy path" do
+    it "supports with POST request for User Update", :vcr do
+      test_data
+
+      user1 = @user1
+
+      expect(user1.username).to eq("username1")
+
+      patch "/api/v0/users/#{user1.id}?username=username0" #, params: { username: "username0" }, headers: {"CONTENT_TYPE" => "application/json", "ACCEPT" => "application/json"}
+
+      expect(response).to be_successful
+      expect(response.status).to eq(200)
+
+      response_data = JSON.parse(response.body, symbolize_names: true)[:data]
+
+      expect(response_data[:type]).to eq("user")
+      expect(response_data[:id]).to be_a(String)
+
+      expect(response_data[:attributes][:email]).to eq("user1@gmail.com")
+      expect(response_data[:attributes][:username]).to eq("username0")
+    end
+  end
+
+  describe "sad path" do
+    it "returns an error if user does not exist", :vcr do
+      test_data
+
+      user1_id = 11
+
+      expect(User.exists?(user1_id)).to be_falsey
+
+      patch "/api/v0/users/#{user1_id}?username=username0" #, params: { username: "username0" }, headers: {"CONTENT_TYPE" => "application/json", "ACCEPT" => "application/json"}
```

Comment:
I think the syntax might be like this (I have not done this in a moment, so is rusty and a few things might be off):
```
user_params = { username: 'username0' }
headers = { 'CONTENT_TYPE' => 'application/json' }
patch "/api/v0/users/#{user1_id}", headers: headers, params: JSON.generate(user: user_params)
```

Reply:
This is similar to what I had before and it is still doing the same thing. The response will be successful, but the username does not change:

```
1) User Update Profile Endpoint happy path supports with POST request for User Update
     Failure/Error: expect(response_data[:attributes][:username]).to eq("username0")
     
       expected: "username0"
            got: "username1"
     
       (compared using ==)
     # ./spec/requests/api/v0/user_requests/update_spec.rb:27:in `block (3 levels) in <top (required)>'
```

Reply:
I actually helped with this 😞 . We did it this way because I know if you use query params, then the json payload will also work, and he was getting a strange error with the normal-ish(?) syntax